mitiq/zne/scaling/tests/test_folding.py

```diff
@@ -1848,3 +1482,255 @@ def test_folding_keeps_measurement_order_with_qiskit():
 
     folded = fold_gates_at_random(circuit, scale_factor=1.0)
     assert folded == circuit
+
+
+def test_create_weight_mask_with_fidelities():
+    """Check the correct weight mask is created."""
+    qreg = LineQubit.range(3)
+    circ = Circuit(
+        ops.H.on_each(*qreg),
+        ops.CNOT.on(qreg[0], qreg[1]),
+        ops.T.on(qreg[2]),
+        ops.TOFFOLI.on(*qreg),
+        ops.measure_each(*qreg),
+    )
+    # Measurement gates should be ignored
+    fidelities = {"H": 0.9, "CNOT": 0.8, "T": 0.7, "TOFFOLI": 0.6}
+    weight_mask = _create_weight_mask(circ, fidelities)
+    assert np.allclose(weight_mask, [0.1, 0.1, 0.1, 0.2, 0.3, 0.4])
+
+    fidelities = {"single": 1.0, "double": 0.5, "triple": 0.1}
+    weight_mask = _create_weight_mask(circ, fidelities)
+    assert np.allclose(weight_mask, [0.0, 0.0, 0.0, 0.5, 0.0, 0.9])
+
+    fidelities = {"single": 1.0, "H": 0.1, "CNOT": 0.2, "TOFFOLI": 0.3}
+    weight_mask = _create_weight_mask(circ, fidelities)
+    assert np.allclose(weight_mask, [0.9, 0.9, 0.9, 0.8, 0.0, 0.7])
+
+
+@pytest.mark.parametrize(
+    "weight_mask",
+    ([0.1, 0.2, 0.3, 0.0], [0.3, 0.5, 0.7, 0.0], [1.0, 1.0, 1.0, 0.0]),
+)
+@pytest.mark.parametrize("scale_factor", (1, 3, 5, 7, 9, 11))
+@pytest.mark.parametrize("method", ("at_random", "from_left", "from_right"))
+def test_create_fold_mask_with_odd_scale_factors(
+    weight_mask, scale_factor, method,
+):
+    """Check _create_fold_mask works as expected"""
```

Comment:
Not that descriptive of a comment, can be inferred from test name. Add more description or remove.